### PR TITLE
Add number of workers to scale up REST API

### LINKED ISSUE #51: [RFE] Ability to specify number of workers to allow REST API to be scaled up
Currently, number of workers are set to 1 in all cases. It would be better to have a configuration option to allow admins to scale up REST API service to use N runners.

* Acceptance criteria
    - new configuration option in configfile
    - documentation
    - configurable workers (see `runner.py` script)

### PR DESCRIPTION
**Pull request by r0x0d**
This patch introduces the ability to increase the number of workers for the REST API via its configuration file.

## Type of change

- [ ] Refactor
- [ ] New feature
- [ ] Bug fix
- [ ] CVE fix
- [ ] Optimization
- [ ] Documentation Update
- [ ] Configuration Update
- [ ] Bump-up dependent library
- [ ] Bump-up library or tool used for development (does not change the final image)
- [ ] CI configuration change
- [ ] Konflux configuration change

## Related Tickets & Documents

- Related Issue #
- Closes #51

## Checklist before requesting a review

- [ ] I have performed a self-review of my code.
- [ ] PR has passed all pre-merge test jobs.
- [ ] If it is a core feature, I have added thorough tests.

## Testing
- Please provide detailed steps to perform tests related to this code change.
- How were the fix/results from this change verified? Please provide relevant screenshots or results.

### CODE REVIEW

**Comment by r0x0d** [comment]
@tisnik probably this still needs some stuff to be finished. I was not sure where the documentation should go, so I found a `config.puml` in the docs folder. Let me know if there is more to be done here and where I should go to finish the task.

File: ols/app/models/config.py [hunk 1]
```diff
@@ -931,6 +931,7 @@ class OLSConfig(BaseModel):
 
     default_provider: Optional[str] = None
     default_model: Optional[str] = None
+    max_workers: Optional[int] = 1
```
Comment (tisnik) [inline ols/app/models/config.py:934]: from lines 934 and 960 it seems it is not `Optional`, just `int`

**Review by tisnik — CHANGES_REQUESTED** [review]
it looks ok, I'd just suggest to remove the "optionality" of this attribute (if pydantic allow it)

**Review by tisnik — APPROVED** [review]
it looks good!

**Pull request merged**